lib/rex/logging/log_dispatcher.rb

```diff
@@ -128,14 +128,40 @@ def get_level(src)
 # still be directed at the correct log file.
 #
 ###
-ExceptionCallStack = "__EXCEPTCALLSTACK__"
+BACKTRACE_LOG_LEVEL = LEV_3
 
 def dlog(msg, src = 'core', level = 0)
   $dispatcher.log(LOG_DEBUG, src, level, msg)
 end
 
-def elog(msg, src = 'core', level = 0)
-  $dispatcher.log(LOG_ERROR, src, level, msg)
+# Logs errors in a standard format for each Log Level.
+#
+# @param msg [String] Contains message from the developer explaining why an error was encountered.
+# Can also be an +Exception+, in which case a log is built from the +Exception+ with no accompanying message.
+#
+# @param src [String] Used to indicate where the error is originating from. Most commonly set to 'core'.
+#
+# @param log_level [Integer] Indicates the level of logging the message should be recorded at. If log_level is greater than
+# the global log level set for +src+, then the log is not recorded.
+#
+# @param error [Exception] Exception of an error that needs to be logged. For all log messages, the class and message of
+# an exception is added to a log message. If the global log level set for +src+ is greater than +BACKTRACE_LOG_LEVEL+,
+# then the stack trace for an error is also added to the log message.
+#
+# (Eg Loop Iterations, Variables, Function Calls).
+#
+# @return [NilClass].
+def elog(msg, src = 'core', log_level = 0, error: nil)
+  global_log_level = get_global_log_level(src)
+  return if log_level > global_log_level
```

Comment:
From what I understand this filtering logic is already handled by the log dispatcher, I think we can safely remove this line:

https://github.com/rapid7/metasploit-framework/blob/ab199e8e4e09f17dafde4954e38898a57c0ce4dc/lib/rex/logging/log_dispatcher.rb#L90-L98

Reply:
Yeah that's fair, the line prevents calling `$dispatcher.log` unnecessarily but there's no need to duplicate logic